File: ios-sdui/HSBCKyc/KYCShellViews.swift
import SwiftUI

// MARK: - Root Navigation View

struct KYCRootView: View {
    @Environment(AppStore.self) private var store

    var body: some View {
        Group {
            if store.isComplete {
                KYCCompletionView()
            } else if store.screenPayload != nil {
                KYCJourneyView()
            } else {
                KYCWelcomeView()
            }
        }
        .animation(.easeInOut(duration: 0.3), value: store.isComplete)
    }
}

// MARK: - Welcome / Start screen

struct KYCWelcomeView: View {
    @Environment(AppStore.self) private var store

    var body: some View {
        ZStack {
            Hive.Color.n50.ignoresSafeArea()

            VStack(spacing: 0) {
                // HSBC Red header bar
                Rectangle()
                    .fill(Hive.Color.brandPrimary)
                    .frame(height: 6)

                ScrollView {
                    VStack(spacing: Hive.Spacing.s6) {
                        // Logo
                        HStack {
                            HSBCLogoView()
                            Spacer()
                        }
                        .padding(.horizontal, Hive.Spacing.s6)
                        .padding(.top, Hive.Spacing.s6)

                        // Hero
                        VStack(alignment: .leading, spacing: Hive.Spacing.s3) {
                            Text("Open Banking\nAccount Opening")
                                .font(Hive.Typography.displayLarge)
                                .foregroundColor(Hive.Color.n800)
                                .lineSpacing(4)

                            Text("Complete your KYC verification in minutes using Open Banking. Fully regulated by the HKMA.")
                                .font(Hive.Typography.bodyBase)
                                .foregroundColor(Hive.Color.n600)
                                .fixedSize(horizontal: false, vertical: true)
                        }
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .padding(.horizontal, Hive.Spacing.s6)

                        // Steps preview
                        VStack(spacing: Hive.Spacing.s3) {
                            ForEach([
                                ("👤", "Personal Details",      "Name, DOB, nationality, address"),
                                ("🪪", "Identity Verification", "Government ID + liveness selfie"),
                                ("💼", "Due Diligence",         "Occupation & source of funds"),
                                ("🏦", "Open Banking",          "Connect bank for instant verification"),
                                ("✍️", "Declarations",          "Legal declarations & consent"),
                            ], id: \.0) { icon, title, desc in
                                HStack(spacing: Hive.Spacing.s4) {
                                    Text(icon).font(.system(size: 24))
                                        .frame(width: 44, height: 44)
                                        .background(Hive.Color.brandPrimaryLight)
                                        .cornerRadius(Hive.Radius.md)
                                    VStack(alignment: .leading, spacing: 2) {
                                        Text(title)
                                            .font(Hive.Typography.labelBase)
                                            .foregroundColor(Hive.Color.n800)
                                        Text(desc)
                                            .font(Hive.Typography.caption)
                                            .foregroundColor(Hive.Color.n500)
                                    }
                                    Spacer()
                                }
                                .padding(Hive.Spacing.s4)
                                .background(Hive.Color.brandWhite)
                                .cornerRadius(Hive.Radius.md)
                                .shadow(color: .black.opacity(0.05), radius: 4, y: 2)
                            }
                        }
                        .padding(.horizontal, Hive.Spacing.s4)

                        // Regulatory note
                        HStack(spacing: Hive.Spacing.s2) {
                            Image(systemName: "lock.shield.fill")
                                .foregroundColor(Hive.Color.n400)
                            Text("Regulated by HKMA · Data protected under PDPO Cap. 486")
                                .font(Hive.Typography.caption)
                                .foregroundColor(Hive.Color.n400)
                        }
                        .padding(.horizontal, Hive.Spacing.s4)

                        // CTA
                        if let err = store.errorMessage {
                            Text(err)
                                .font(Hive.Typography.bodySm)
                                .foregroundColor(Hive.Color.error)
                                .multilineTextAlignment(.center)
                                .padding(.horizontal, Hive.Spacing.s6)
                        }

                        Button("Begin Application") {
                            store.dispatch(.startSession)
                        }
                        .buttonStyle(HivePrimaryButtonStyle(isLoading: store.isLoading))
                        .disabled(store.isLoading)
                        .padding(.horizontal, Hive.Spacing.s4)
                        .padding(.bottom, Hive.Spacing.s8)
                    }
                }
            }
        }
    }
}

// MARK: - Journey shell (progress bar + step content)

struct KYCJourneyView: View {
    @Environment(AppStore.self) private var store

    var body: some View {
        VStack(spacing: 0) {
            // Header
            KYCHeader()

            // Progress bar
            KYCMobileProgressBar()

            // Step content (scrollable)
            // .id on ScrollView resets scroll position and forces full re-render on step change
            ScrollView {
                VStack(alignment: .leading, spacing: Hive.Spacing.s5) {
                    // Section badge + step title
                    VStack(alignment: .leading, spacing: Hive.Spacing.s2) {
                        HiveSectionBadge(label: store.sectionTitle)
                        if let payload = store.screenPayload {
                            KYCStepTitleView(payload: payload)
                        }
                    }

                    // SDUI-rendered step content — .id inside KYCStepContentCard
                    // forces destroy/recreate of step views on stepId change
                    if let payload = store.screenPayload {
                        KYCStepContentCard(payload: payload)
                    }
                }
                .padding(.horizontal, Hive.Spacing.s4)
                .padding(.top, Hive.Spacing.s5)
                .padding(.bottom, 110)
            }
            .id(store.currentStepId ?? "loading") // resets scroll + forces re-render
            .background(Hive.Color.n50)

            // Sticky nav bar
            KYCNavBar()
        }
        .background(Hive.Color.n50)
        .ignoresSafeArea(edges: .bottom)
        .overlay(
            Group {
                if store.isLoading {
                    KYCLoadingOverlay()
                }
            }
        )
    }
}

// MARK: - Header

struct KYCHeader: View {
    @Environment(AppStore.self) private var store

    var body: some View {
        HStack {
            HSBCLogoView()
            Spacer()
            Button("Save & Exit") { store.dispatch(.setError(nil)) }
                .font(Hive.Typography.bodySm)
                .foregroundColor(Hive.Color.n500)
        }
        .padding(.horizontal, Hive.Spacing.s4)
        .padding(.vertical, Hive.Spacing.s3)
        .background(Hive.Color.brandWhite)
        .overlay(
            Rectangle()
                .frame(height: 1)
                .foregroundColor(Hive.Color.n200),
            alignment: .bottom
        )
    }
}

// MARK: - Progress bar

struct KYCMobileProgressBar: View {
    @Environment(AppStore.self) private var store

    var progress: CGFloat {
        guard store.totalSteps > 0 else { return 0 }
        return CGFloat(store.currentStepIndex) / CGFloat(store.totalSteps)
    }

    var body: some View {
        VStack(spacing: Hive.Spacing.s2) {
            HStack {
                Text(store.sectionTitle)
                    .font(Hive.Typography.caption)
                    .foregroundColor(Hive.Color.n500)
                Spacer()
                Text("Step \(store.currentStepIndex) of \(store.totalSteps)")
                    .font(Hive.Typography.caption)
                    .foregroundColor(Hive.Color.n500)
            }

            GeometryReader { geo in
                ZStack(alignment: .leading) {
                    Capsule()
                        .fill(Hive.Color.n200)
                        .frame(height: 4)
                    Capsule()
                        .fill(Hive.Color.brandPrimary)
                        .frame(width: geo.size.width * progress, height: 4)
                        .animation(.easeInOut(duration: 0.4), value: progress)
                }
            }
            .frame(height: 4)
        }
        .padding(.horizontal, Hive.Spacing.s4)
        .padding(.vertical, Hive.Spacing.s3)
        .background(Hive.Color.brandWhite)
        .overlay(
            Rectangle().frame(height: 1).foregroundColor(Hive.Color.n200),
            alignment: .bottom
        )
    }
}

// MARK: - Step title

struct KYCStepTitleView: View {
    let payload: SDUIScreenPayload

    private var stepIcon: String {
        let icons = [
            "name":"👤", "dob":"🎂", "contact":"📱", "identifier":"🪪",
            "address":"🏠", "document":"📄", "liveness":"😊",
            "wealth":"💼", "openbanking":"🏦", "declaration":"✍️",
        ]
        for (key, icon) in icons {
            if payload.metadata.stepId.contains(key) { return icon }
        }
        return "📋"
    }

    private var stepTitle: String {
        let titles = [
            "name":"Full Legal Name", "dob":"Date of Birth & Nationality",
            "contact":"Contact Details", "identifier":"Government ID Number",
            "address":"Residential Address", "document":"Upload Identity Document",
            "liveness":"Selfie & Liveness Check", "wealth":"Occupation & Source of Funds",
            "openbanking":"Connect Your Bank", "declaration":"Legal Declarations",
        ]
        for (key, title) in titles {
            if payload.metadata.stepId.contains(key) { return title }
        }
        return payload.metadata.sectionTitle
    }

    var body: some View {
        Text("\(stepIcon) \(stepTitle)")
            .font(Hive.Typography.headingMd)
            .foregroundColor(Hive.Color.n800)
            .fixedSize(horizontal: false, vertical: true)
    }
}

// MARK: - Step content card

struct KYCStepContentCard: View {
    let payload: SDUIScreenPayload

    private var questionIds: [String] {
        payload.layout.children?
            .compactMap { $0.props?["questionId"]?.stringValue } ?? []
    }

    var body: some View {
        VStack(alignment: .leading, spacing: Hive.Spacing.s5) {
            KYCSDUIStepRouter(
                stepId: payload.metadata.stepId,
                sectionTitle: payload.metadata.sectionTitle,
                questionIds: questionIds
            )
        }
        .padding(Hive.Spacing.s5)
        .background(Hive.Color.brandWhite)
        .cornerRadius(Hive.Radius.lg)
        .shadow(color: .black.opacity(0.06), radius: 8, y: 2)
        // CRITICAL: force SwiftUI to destroy and recreate the entire step
        // view tree when the stepId changes, so @State inside each step
        // (TextFields, Pickers, etc.) is reset and routed to the new view type
        .id(payload.metadata.stepId)
    }
}

// MARK: - Navigation bar

struct KYCNavBar: View {
    @Environment(AppStore.self) private var store

    var isFirstStep: Bool { store.currentStepIndex <= 1 }
    var isLastStep:  Bool { store.currentStepIndex >= store.totalSteps }

    var body: some View {
        HStack(spacing: Hive.Spacing.s3) {
            if !isFirstStep {
                Button(action: {
                    let prevIdx = store.currentStepIndex - 1
                    guard prevIdx >= 1, let sid = store.sessionId else { return }
                    let prevStepId = String(format: "step-%03d", prevIdx)
                    store.dispatch(.setLoading(true))
                    Task {
                        let net = KYCNetworkService()
                        do {
                            let payload = try await net.getStep(sessionId: sid, stepId: prevStepId, platform: "mobile")
                            await MainActor.run { store.dispatch(.stepLoaded(payload)) }
                        } catch {
                            await MainActor.run { store.dispatch(.setError("Could not go back.")) }
                        }
                    }
                }) {
                    HStack(spacing: Hive.Spacing.s1) {
                        Image(systemName: "chevron.left")
                        Text("Back")
                    }
                    .font(Hive.Typography.bodyBase)
                    .foregroundColor(Hive.Color.n700)
                    .padding(.horizontal, Hive.Spacing.s4)
                    .frame(height: Hive.Component.Button.height)
                    .background(Hive.Color.brandWhite)
                    .cornerRadius(Hive.Radius.base)
                    .overlay(
                        RoundedRectangle(cornerRadius: Hive.Radius.base)
                            .stroke(Hive.Color.n300, lineWidth: 1)
                    )
                }
            }

            Button(action: { store.dispatch(.submitStep) }) {
                Text(isLastStep ? "Submit Application" : "Continue")
            }
            .buttonStyle(HivePrimaryButtonStyle(isLoading: store.isSubmitting))
            .disabled(store.isSubmitting)
        }
        .padding(.horizontal, Hive.Spacing.s4)
        .padding(.top, Hive.Spacing.s4)
        .padding(.bottom, Hive.Spacing.s6)
        .background(Hive.Color.brandWhite)
        .overlay(
            Rectangle().frame(height: 1).foregroundColor(Hive.Color.n200),
            alignment: .top
        )
        .shadow(color: .black.opacity(0.06), radius: 8, y: -2)
    }
}

// MARK: - Loading overlay

struct KYCLoadingOverlay: View {
    var body: some View {
        ZStack {
            Color.black.opacity(0.3).ignoresSafeArea()
            VStack(spacing: Hive.Spacing.s4) {
                ProgressView()
                    .progressViewStyle(CircularProgressViewStyle(tint: Hive.Color.brandPrimary))
                    .scaleEffect(1.4)
                Text("Loading…")
                    .font(Hive.Typography.bodySm)
                    .foregroundColor(Hive.Color.n700)
            }
            .padding(Hive.Spacing.s8)
            .background(Hive.Color.brandWhite)
            .cornerRadius(Hive.Radius.lg)
            .shadow(color: .black.opacity(0.15), radius: 16)
        }
    }
}

// MARK: - Completion screen

struct KYCCompletionView: View {
    var body: some View {
        ZStack {
            Hive.Color.n50.ignoresSafeArea()
            VStack(spacing: Hive.Spacing.s6) {
                HSBCLogoView()
                Text("✅").font(.system(size: 72))
                Text("Application Submitted")
                    .font(Hive.Typography.headingXL)
                    .foregroundColor(Hive.Color.success)
                    .multilineTextAlignment(.center)

                VStack(alignment: .leading, spacing: Hive.Spacing.s3) {
                    ForEach([
                        ("📧", "Confirmation email sent"),
                        ("📱", "SMS notification will follow"),
                        ("⏱",  "Decision within 3 working days"),
                        ("🔒", "Data encrypted per PDPO Cap. 486"),
                    ], id: \.0) { icon, text in
                        HStack(spacing: Hive.Spacing.s3) {
                            Text(icon)
                            Text(text)
                                .font(Hive.Typography.bodySm)
                                .foregroundColor(Hive.Color.n700)
                        }
                    }
                }
                .padding(Hive.Spacing.s5)
                .background(Hive.Color.n50)
                .cornerRadius(Hive.Radius.md)
                .overlay(
                    RoundedRectangle(cornerRadius: Hive.Radius.md)
                        .stroke(Hive.Color.n200, lineWidth: 1)
                )

                Text("Reference: OBKYC-\(String(Int.random(in: 100000...999999)))")
                    .font(Hive.Typography.caption)
                    .foregroundColor(Hive.Color.n400)
            }
            .padding(Hive.Spacing.s6)
        }
    }
}

// MARK: - HSBC Logo

struct HSBCLogoView: View {
    var body: some View {
        HStack(spacing: Hive.Spacing.s2) {
            // HSBC hexagon mark
            ZStack {
                Grid(alignment: .topLeading, horizontalSpacing: 0, verticalSpacing: 0) {
                    GridRow {
                        Rectangle().fill(Hive.Color.brandPrimary).frame(width: 14, height: 14)
                        Rectangle().fill(Hive.Color.brandWhite).frame(width: 14, height: 14)
                    }
                    GridRow {
                        Rectangle().fill(Hive.Color.brandWhite).frame(width: 14, height: 14)
                        Rectangle().fill(Hive.Color.brandPrimary).frame(width: 14, height: 14)
                    }
                }
            }
            .frame(width: 28, height: 28)
            .overlay(Rectangle().stroke(Hive.Color.n200, lineWidth: 0.5))

            Text("HSBC")
                .font(.system(size: 20, weight: .bold))
                .foregroundColor(Hive.Color.n800)
                .kerning(1)
        }
    }
}

File: android-sdui/app/src/main/java/com/hsbc/sdui/kyc/KYCShellViews.kt
package com.hsbc.sdui.kyc

import androidx.compose.animation.AnimatedContent
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.togetherWith
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel

private val HsbcRed = Color(0xFFC41230)
private val N50     = Color(0xFFF8F8F8)
private val N200    = Color(0xFFE5E5E5)
private val N500    = Color(0xFF666666)
private val N700    = Color(0xFF333333)
private val N800    = Color(0xFF1A1A1A)
private val N900    = Color(0xFF0A0A0A)
private val White   = Color.White
private val Success = Color(0xFF2D7D46)

// ─── Root screen entry point ──────────────────────────────────────────────────

@Composable
fun KYCRootScreen(vm: KYCViewModel = viewModel()) {
    val state by vm.state.collectAsState()

    AnimatedContent(
        targetState = state.isComplete to (state.screenPayload != null),
        transitionSpec = { fadeIn() togetherWith fadeOut() },
        label = "KYCRoot"
    ) { (isComplete, hasPayload) ->
        when {
            isComplete  -> KYCCompletionScreen()
            hasPayload  -> KYCJourneyScreen(state, vm::setAnswer, vm::submitStep)
            else        -> KYCWelcomeScreen(state, vm::startSession)
        }
    }
}

// ─── Welcome screen ───────────────────────────────────────────────────────────

@Composable
private fun KYCWelcomeScreen(state: KYCUiState, onStart: () -> Unit) {
    Box(modifier = Modifier.fillMaxSize().background(N50)) {
        Column(modifier = Modifier.verticalScroll(rememberScrollState())) {
            // HSBC red accent bar
            Box(modifier = Modifier.fillMaxWidth().height(6.dp).background(HsbcRed))

            Column(modifier = Modifier.padding(24.dp), verticalArrangement = Arrangement.spacedBy(24.dp)) {
                HSBCLogo()

                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    Text("Open Banking\nAccount Opening",
                        fontSize = 28.sp, fontWeight = FontWeight.Bold, color = N900, lineHeight = 36.sp)
                    Text("Complete your KYC verification in minutes using Open Banking. Fully regulated by the HKMA.",
                        fontSize = 15.sp, color = Color(0xFF555555))
                }

                Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                    listOf(
                        Triple("👤", "Personal Details", "Name, DOB, nationality, address"),
                        Triple("🪪", "Identity Verification", "Government ID + liveness selfie"),
                        Triple("💼", "Due Diligence", "Occupation & source of funds"),
                        Triple("🏦", "Open Banking", "Connect bank for instant verification"),
                        Triple("✍️", "Declarations", "Legal declarations & consent")
                    ).forEach { (icon, title, desc) ->
                        Card(
                            colors = CardDefaults.cardColors(containerColor = White),
                            modifier = Modifier.fillMaxWidth()
                        ) {
                            Row(Modifier.padding(16.dp), verticalAlignment = Alignment.CenterVertically) {
                                Box(
                                    modifier = Modifier.size(44.dp)
                                        .clip(RoundedCornerShape(8.dp))
                                        .background(Color(0xFFFFEEF0)),
                                    contentAlignment = Alignment.Center
                                ) { Text(icon, fontSize = 22.sp) }
                                Spacer(Modifier.width(12.dp))
                                Column {
                                    Text(title, fontWeight = FontWeight.Medium, fontSize = 14.sp, color = N800)
                                    Text(desc, fontSize = 12.sp, color = N500)
                                }
                            }
                        }
                    }
                }

                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text("🔒", fontSize = 14.sp)
                    Spacer(Modifier.width(6.dp))
                    Text("Regulated by HKMA · Data protected under PDPO Cap. 486",
                        fontSize = 12.sp, color = N500)
                }

                state.errorMessage?.let {
                    Text(it, fontSize = 13.sp, color = Color(0xFFCC0000))
                }

                Button(
                    onClick = onStart,
                    enabled = !state.isLoading,
                    modifier = Modifier.fillMaxWidth().height(52.dp),
                    shape = RoundedCornerShape(8.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = HsbcRed)
                ) {
                    if (state.isLoading) {
                        CircularProgressIndicator(color = White, modifier = Modifier.size(20.dp), strokeWidth = 2.dp)
                    } else {
                        Text("Begin Application", fontWeight = FontWeight.SemiBold, fontSize = 16.sp, color = White)
                    }
                }

                Spacer(Modifier.height(24.dp))
            }
        }
    }
}

// ─── Journey shell ────────────────────────────────────────────────────────────

@Composable
private fun KYCJourneyScreen(
    state: KYCUiState,
    onAnswer: (String, Any) -> Unit,
    onSubmit: () -> Unit
) {
    val payload = state.screenPayload ?: return
    val stepId = payload.metadata.stepId
    val progress = if (state.totalSteps > 0)
        state.currentStepIndex.toFloat() / state.totalSteps.toFloat() else 0f

    Scaffold(
        topBar = {
            Column {
                // Header
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(White)
                        .padding(horizontal = 16.dp, vertical = 12.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    HSBCLogo()
                    Text("Save & Exit", fontSize = 13.sp, color = N500)
                }
                Divider(color = N200)

                // Progress bar
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(White)
                        .padding(horizontal = 16.dp, vertical = 10.dp)
                ) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Text(state.sectionTitle, fontSize = 12.sp, color = N500)
                        Text("Step ${state.currentStepIndex} of ${state.totalSteps}",
                            fontSize = 12.sp, color = N500)
                    }
                    Spacer(Modifier.height(6.dp))
                    LinearProgressIndicator(
                        progress = { progress },
                        modifier = Modifier.fillMaxWidth().height(4.dp)
                            .clip(RoundedCornerShape(2.dp)),
                        color = HsbcRed,
                        trackColor = N200
                    )
                }
                Divider(color = N200)
            }
        },
        bottomBar = {
            Column {
                Divider(color = N200)
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(White)
                        .padding(horizontal = 16.dp, vertical = 12.dp),
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    if (state.currentStepIndex > 1) {
                        OutlinedButton(
                            onClick = {},
                            modifier = Modifier.height(52.dp),
                            shape = RoundedCornerShape(8.dp)
                        ) {
                            Text("← Back", color = N700)
                        }
                    }
                    Button(
                        onClick = onSubmit,
                        enabled = !state.isSubmitting,
                        modifier = Modifier.weight(1f).height(52.dp),
                        shape = RoundedCornerShape(8.dp),
                        colors = ButtonDefaults.buttonColors(containerColor = HsbcRed)
                    ) {
                        if (state.isSubmitting) {
                            CircularProgressIndicator(color = White, modifier = Modifier.size(20.dp), strokeWidth = 2.dp)
                        } else {
                            Text(
                                if (state.currentStepIndex >= state.totalSteps) "Submit Application" else "Continue",
                                fontWeight = FontWeight.SemiBold, color = White
                            )
                        }
                    }
                }
            }
        }
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(N50)
                .padding(innerPadding)
                .verticalScroll(rememberScrollState())
        ) {
            Column(modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(16.dp)) {
                // Section badge + title
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    Box(
                        modifier = Modifier
                            .clip(RoundedCornerShape(20.dp))
                            .background(Color(0xFFFFEEF0))
                            .padding(horizontal = 12.dp, vertical = 4.dp)
                    ) {
                        Text(state.sectionTitle,
                            fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = HsbcRed)
                    }
                    Text(kycStepTitle(primaryQuestionId(payload)),
                        fontSize = 20.sp, fontWeight = FontWeight.Bold, color = N900)
                }

                // Step content card
                Card(
                    colors = CardDefaults.cardColors(containerColor = White),
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(12.dp),
                    elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
                ) {
                    Column(Modifier.padding(16.dp)) {
                        KYCStepRouter(
                            primaryQuestionId = primaryQuestionId(payload),
                            stepId = stepId,
                            answers = state.answers,
                            onAnswer = onAnswer
                        )
                    }
                }

                state.errorMessage?.let {
                    Text(it, fontSize = 13.sp, color = Color(0xFFCC0000))
                }

                Spacer(Modifier.height(8.dp))
            }
        }

        if (state.isLoading) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(Color.Black.copy(alpha = 0.3f)),
                contentAlignment = Alignment.Center
            ) {
                Card(shape = RoundedCornerShape(12.dp)) {
                    Column(
                        modifier = Modifier.padding(32.dp),
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        CircularProgressIndicator(color = HsbcRed)
                        Spacer(Modifier.height(12.dp))
                        Text("Loading…", fontSize = 14.sp, color = N700)
                    }
                }
            }
        }
    }
}


/** Extract primary question ID from SDUI payload — mirrors iOS KYCStepContentCard.questionIds */
fun primaryQuestionId(payload: SDUIScreenPayload): String {
    val children = payload.layout.children ?: return ""
    val first = children.firstOrNull() ?: return ""
    val props = first.props ?: return ""
    return props["questionId"] as? String ?: ""
}

// ─── Completion screen ────────────────────────────────────────────────────────

@Composable
private fun KYCCompletionScreen() {
    Box(
        modifier = Modifier.fillMaxSize().background(N50).padding(24.dp),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(20.dp)
        ) {
            HSBCLogo()
            Text("✅", fontSize = 72.sp)
            Text("Application Submitted",
                fontSize = 24.sp, fontWeight = FontWeight.Bold, color = Success)

            Card(
                colors = CardDefaults.cardColors(containerColor = N50),
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(12.dp)
            ) {
                Column(Modifier.padding(20.dp), verticalArrangement = Arrangement.spacedBy(10.dp)) {
                    listOf(
                        "📧" to "Confirmation email sent",
                        "📱" to "SMS notification will follow",
                        "⏱" to "Decision within 3 working days",
                        "🔒" to "Data encrypted per PDPO Cap. 486"
                    ).forEach { (icon, text) ->
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Text(icon, fontSize = 16.sp)
                            Spacer(Modifier.width(12.dp))
                            Text(text, fontSize = 14.sp, color = N700)
                        }
                    }
                }
            }

            Text(
                "Reference: OBKYC-${(100000..999999).random()}",
                fontSize = 12.sp, color = N500
            )
        }
    }
}

// ─── HSBC Logo ────────────────────────────────────────────────────────────────

@Composable
private fun HSBCLogo() {
    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
        Box(modifier = Modifier.size(28.dp).border(0.5.dp, N200)) {
            Column {
                Row {
                    Box(Modifier.size(14.dp).background(HsbcRed))
                    Box(Modifier.size(14.dp).background(White))
                }
                Row {
                    Box(Modifier.size(14.dp).background(White))
                    Box(Modifier.size(14.dp).background(HsbcRed))
                }
            }
        }
        Text("HSBC", fontSize = 20.sp, fontWeight = FontWeight.Bold, color = N900,
            letterSpacing = 1.sp)
    }
}
